lib/ae_mdw/sync/async_tasks/store.ex

```diff
@@ -53,44 +60,39 @@ defmodule AeMdw.Sync.AsyncTasks.Store do
 
   @spec set_processing(task_index()) :: :ok
   def set_processing(task_index) do
-    :ets.insert(:async_tasks_processing, {task_index})
+    :ets.insert(@processing_tab, {task_index})
     :ok
   end
 
-  @spec set_done(task_index()) :: :ok
-  def set_done(task_index) do
+  @spec set_done(task_index(), task_args()) :: :ok
+  def set_done({_ts, task_type} = task_index, args) do
     :mnesia.sync_transaction(fn ->
       :mnesia.delete(Model.AsyncTasks, task_index, :write)
     end)
 
-    :ets.delete_object(:async_tasks_processing, task_index)
+    :ets.delete_object(@processing_tab, task_index)
+    :ets.delete(@args_tab, {task_type, args})
     :ok
   end
 
   #
   # Private functions
   #
-  defp is_enqueued?(task_type, args) do
-    exists_spec =
-      Ex2ms.fun do
-        {:_, {:_, ^task_type}, ^args} -> true
-      end
+  defp cache_tasks_by_args() do
+    {:atomic, indexed_args_records} =
+      :mnesia.transaction(fn ->
+        args_spec =
+          Ex2ms.fun do
+            Model.async_tasks(index: {_ts, task_type}, args: args) -> {{task_type, args}}
+          end
+
+        :mnesia.select(Model.AsyncTasks, args_spec)
```

Comment:
Wouldn't it be more efficient here if we used `:mnesia.all_keys` instead, and then map every value to obtain the `{task_type, args}` tuples?

Reply:
profiled now ... `:mnesia.select` too only 100ms for 10k items (runs once on app start).